website/src/components/canvas/sidebar/ResizeTextfields.jsx

```diff
@@ -0,0 +1,172 @@
+/**
+ * @file Component with textfields to resize canvas elements by height and width
+ */
+
+import React, {PropTypes} from 'react';
+import {connect} from 'react-redux';
+
+import AddCircle from 'material-ui/svg-icons/content/add-circle';
+import IconButton from 'material-ui/IconButton';
+import RemoveCircle from 'material-ui/svg-icons/content/remove-circle';
+import TextField from 'material-ui/TextField';
+
+import * as AC from '../../../redux/actions/ActionConstants';
+import * as EA from '../../../redux/actions/ElementActions';
+
+const styles = {
+    field: {
+        width: '30%'
+    },
+    smallIcon: {
+        height: 36,
+        width: 36
+    },
+    small: {
+        display: 'inline-block',
+        height: 60,
+        padding: 2,
+        width: 60
+    }
+};
+
+/**
+* Component of reisze textfields
+*/
+class ResizeTextfields extends React.Component {
+    /**
+   * Constructor for ResizeTextfields
+   * @param {Object} props The props for the ResizeTextfields.
+   */
+    constructor(props) {
+        super(props);
+        this.handleButtonAddHeight = this.handleButtonAddHeight.bind(this);
+        this.handleButtonAddWidth = this.handleButtonAddWidth.bind(this);
+        this.handleButtonSubHeight = this.handleButtonSubHeight.bind(this);
+        this.handleButtonSubWidth = this.handleButtonSubWidth.bind(this);
+        this.handlerTextfieldHeight = this.handlerTextfieldHeight.bind(this);
+        this.handlerTextfieldWidth = this.handlerTextfieldWidth.bind(this);
+    }
+    /**
+     * Handler that adds 1px to height
+     * @returns {void}
+     */
+    handleButtonAddHeight() {
```

Comment:
Instead of hardcoding these four functions might be nicer to have something like the following:

```
generateButtonHandler(changeAmount, widthOrHeight) {
  const handler = () => {
    const newSize = Object.assign(this.props.elements[this.props.targetedId]);
    newSize[widthOrHeight] += changeAmount;
    this.props.dispatch(EA.updateAndPersist(AC.UPDATE_SIZE,
      this.props.targetedId, newSize, this.props.currentCanvas));
  }
  return handler;
}
```

Then when you are attaching the handler you would do...

```
onTouchTap={this.generateButtonHandler(-1 * UNIT_CHANGE, WIDTH)}
```